build-llvm.py

```diff
@@ -68,6 +68,17 @@ def parse_parameters(root_folder):
                         type=str,
                         default=os.path.join(root_folder.as_posix(), "build",
                                              "llvm"))
+    parser.add_argument("--clang-vendor",
+                        help=textwrap.dedent("""\
+                        Add this value to the clang version string (like "Apple clang version..." or
+                        "Android clang version..."). Useful when reverting or applying patches on top
+                        of upstream clang to differentiate a toolchain built with this script from
+                        upstream clang or to distinguish a toolchain built with this script from the
+                        system's clang. Defaults to empty.
+
+                        """),
+                        type=str,
+                        default="")
```

Comment:
Isn't it easier to default to something special here, since you know that this is the ClangBuiltLinux script (rather than altering a bunch of different build configs)?

Reply:
I suppose it could be `ClangBuiltLinux` but by default, we are just building upstream clang so it seems a little weird to add a vendor in that case. That's just me personally though, I can make that change if you feel strongly about it!

Reply:
I see what you are saying, but this isn't upstream Clang. You are configuring it to only have certain backends and other options. An official upstream build would have all the targets available. I don't particularly feel strongly about this, but I thought that the thing the other users would want is a way to know that this script definitely built their Clang.

Reply:
+1 on setting a `ClangBuiltLinux` default.

Reply:
Done: https://github.com/nathanchance/tc-build/commit/03967cf6bcec7ce00fc476f7cfcfa03661438be4